beps/bep_0003.rst

```diff
@@ -380,6 +429,24 @@ decent chance of getting a complete piece to upload, new connections
 are three times as likely to start as the current optimistic unchoke
 as anywhere else in the rotation.
 
+
+------------
+Upgrade Path
+------------
+
+## TODO ##
+
+* restrict file layout. no nested directories for hybrid torrents
+* padding or different piece-space layout?
+* pieces field
+* double announce behavior
+* safe hashing. avoid downgrade attacks
+* changes to BEP 9. magnets. send merkle layers.
```

Comment:
interesting. I imagine the main value from merkle trees is to extend the PIECE message to include uncle-hashes (like the tribler protocol does). But I suppose extending BEP 9 may provide a simpler upgrade path.